### PR TITLE
[Fairground 🎡] Expanded nav alignment and styling

### PR DESCRIPTION
**Pull request by frederickobrien**
Following on from #11572, this adjusts the alignment and styling of the new masthead expanded nav components to _align_ (eh? eh?) with new designs. 

The components are not reactive with this implementation (i.e. the expanded nav is constantly in the expanded state) but it looks as it should in that state. With a delicious new veggie burger all the pieces will be combined.

Some things to note:

- The search bar has been extracted into its own component to allow for it to appear in two different places across mobile-to-tablet and desktop-plus
- Colours have be routed through the `palette.ts` hub rather than being imported directly from Source 

What this does **not** do:

- Interactive veggie burger behaviour

To be followed up with another PR in the chain.

### Desktop

image

### LeftCol

image

### Wide

image

### CODE REVIEW

File: dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNav.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+//         to apply styles to the nested tabs due to the fact we use ~
+//         to support NoJS
+//     */
+// 	/* stylelint-disable-next-line selector-type-no-unknown */
+// 	${`#${navInputCheckboxId}`}:checked ~ div & {
+// 		${from.desktop} {
+// 			display: block;
+// 			overflow: visible;
+// 		}
+// 	}
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNav.tsx:32]: Are these comment left over for a specific reason?

File: dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavColumn.tsx [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+			<script
+				dangerouslySetInnerHTML={{
+					__html: `document.addEventListener('DOMContentLoaded', function(){
+                        var columnInput = document.getElementById('${collapseColumnInputId}');
+
+						if (!columnInput) return; // Sticky nav replaces the nav so element no longer exists for users in test.
+
+						columnInput.addEventListener('keydown', function(e){
+                            // keyCode: 13 => Enter key | keyCode: 32 => Space key
+                            if (e.keyCode === 13 || e.keyCode === 32) {
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavColumn.tsx:253]: FYI [`keyCode` is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) in favour of [`code`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code)

File: dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavColumn.tsx [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+`;
+
+const columnLinkTitle = css`
+	${textSans15};
+	/**
+	 * Typography preset styles should not be overridden.
+	 * This has been done because the styles do not directly map to the new presets.
+	 * Please speak to your team's designer and update this to use a more appropriate preset.
+	 */
+	line-height: 1.15;
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavColumn.tsx:47]: Do you plan on addressing these inconsitencies as part of this work?

File: dotcom-rendering/src/components/Masthead/Titlepiece/Titlepiece.tsx [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+					dataLinkName={''}
+				/>
+			</div>
+
+			{/* Guardian logo */}
+			<div css={guardianLogoStyles}>
+				<a href="https://theguardian.com">
+					<SvgGuardianLogo
+						textColor={themePalette('--masthead-nav-link-text')}
+					/>
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Masthead/Titlepiece/Titlepiece.tsx:173]: Are you aware that for specific editions at specific times the logo changes?

**Review by mxdvl — COMMENTED** [review]
Did you mean to request a review from @guardian/dotcom-platform specifically? Is there a place where we should focus our attention? I’ve left a few questions and notes, but unsure if they are helpful or only on existing things copied in a new place…

File: dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavSections.tsx [hunk 5]
```diff
@@ -1,46 +1,37 @@
...
 import { ReaderRevenueLinks } from './ExpandedNavReaderRevenueLinks';
+import { ExpandedNavSearchBar } from './ExpandedNavSearchBar';
 
 const columnsStyle = (isImmersive: boolean) => css`
 	box-sizing: border-box;
 	max-width: none;
 	${from.desktop} {
 		max-width: 980px;
-		padding: 0 20px;
+		padding: 0 ${space[4] + 3}px;
```
[the remark this answers is not in the document]
Comment (frederickobrien) [reply dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavSections.tsx]: Ah that's my bad, I was eyeballing the padding at the bottom of the expanded nav. I'll switch to a fully Source space

[the remark this answers is not in the document]
Comment (frederickobrien) [reply dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavSections.tsx]: Ah there's a tricky alignment issue that I hadn't gotten to the bottom of. This gets the main pillars aligned with the expanded ones but obviously not a sustainable solution

File: dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavSearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+	const searchId = 'gu-search-mobile';
+	return (
+		<form css={searchBar} action="https://www.google.co.uk/search">
+			<TextInput
+				hideLabel={true}
+				label="Search input"
+				cssOverrides={searchInput}
+				name="q" // query param sent to google
+				placeholder="Search the Guardian"
+				data-link-name={nestedOphanComponents('nav2', 'search')}
```
[the remark this answers is not in the document]
Comment (frederickobrien) [reply dotcom-rendering/src/components/Masthead/Titlepiece/ExpandedNav/ExpandedNavSearchBar.tsx]: Addressed with c96995513195809aa4c817daca1bb9a232be7822

**Pull request merged**